### PR TITLE
ENH: Improve CI.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Improve the module's CI:
- Add Circle CI `*.yml` file.
- Add the `CTestConfig.cmake` file to submit builds to the **Insight**
  project in the **open.cdash.org** dashboard.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Not sure whether this requires any wrapping/whether it makes sense here: I added an empty `CMakeLists.txt` file. @phcerdan @thewtex please comment on this. Thanks !

**Comment by thewtex** [comment]
Thanks, @jhlegarreta 

We can skip wrapping for this module.

**Comment by jhlegarreta** [comment]
Done in c189d20.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            BUILD_TESTING:BOOL=ON
+            ")
+            include(\${CTEST_SCRIPT_DIRECTORY}/itk_common.cmake)
+            EOF
+      - run:
+          name: Build and Test
+          no_output_timeout: 1.0h
+          command: |
+            ctest -j 2 -VV -S dashboard.cmake
+  package:
```
Comment (thewtex) [inline .circleci/config.yml]: We can remove this section, since we are not building a package.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+sudo: true
```
Comment (thewtex) [inline .travis.yml]: We can remove this file, since we are not building a package.

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+branches:
```
Comment (thewtex) [inline appveyor.yml]: We can remove this file, since we are not building a package.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
```
Comment (thewtex) [inline setup.py]: We can remove this file, since we are not building a package.

**Review by thewtex — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (jhlegarreta) [reply .circleci/config.yml]: Done in cf1093a. Thanks Matt.

File: .travis.yml (hunk 2 above)
Comment (jhlegarreta) [reply .travis.yml]: Done in cf1093a. Thanks Matt.

File: setup.py (hunk 4 above)
Comment (jhlegarreta) [reply setup.py]: Done in cf1093a. Thanks Matt.

**Review by thewtex — COMMENTED** [review]
Thanks for the updates, @jhlegarreta 

I disabled the TravisCI and Appveyor builds.

A `KWStyleOverwrite.txt` could be added to address the current test failure.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+          no_output_timeout: 1.0h
+          command: |
+            ctest -j 2 -VV -S dashboard.cmake
+
+workflows:
+    version: 2
+    build-test-package:
+      jobs:
+        - build-and-test
+        - package
```
Comment (thewtex) [inline .circleci/config.yml]: The reference to the `package` job can be removed.

**Review by thewtex — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 5 above)
Comment (jhlegarreta) [reply .circleci/config.yml]: Done in ab5b15e.

**Comment by jhlegarreta** [comment]
File `ITKKWStyleOverwrite.txt` added in #5.

**Review by thewtex — APPROVED** [review]

**Pull request merged**